cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -377,7 +391,12 @@ public void follower() {
           LOG.info("Stopping master twill application");
           TwillController twillController = controller.get();
           if (twillController != null) {
-            twillController.stopAndWait();
+            try {
+              twillController.terminate();
+              twillController.awaitTerminated();
```

Comment:
`terminate()` returns a `Future`, so you can just do `Futures.getUnchecked(twillController.terminate())` to achieve the same effect as before.